src/liballoc/arc.rs

```diff
@@ -282,6 +282,9 @@ impl<T> Arc<T> {
     /// To avoid a memory leak the pointer must be converted back to an `Arc` using
     /// [`Arc::from_raw`][from_raw].
     ///
+    /// If `T` is zero-sized (e.g. `Arc<()>`), the returned pointer address
+    /// will be meaningless.
```

Comment:
The value is one-byte-past-the-refcounts and can be used to get back an `Rc`, and you can distinguish between them, so definitely not meaningless (whereas an `&ZST` by itself is useless for anything short of a proof that there is a value of type `ZST` somewhere).